src/super_gradients/training/sg_model/sg_model.py

```diff
@@ -320,6 +313,87 @@ def build_model(self,  # noqa: C901 - too complex
+                       student_arch_params={},
+                       teacher_arch_params={},
+                       teacher_checkpoint_path: str = None,
+                       load_kd_model_checkpoint: bool = False,
+                       kd_model_source_ckpt_folder_name: str = None,
+                       kd_model_external_checkpoint_path: str = None,
+                       freeze_teacher_eval_mode: bool = False,
+                       ):
+        """
+        :param student_architecture:       Defines the student's architecture from models/ALL_ARCHITECTURES
+        :param teacher_architecture:       Defines the teacher's architecture from models/ALL_ARCHITECTURES
+        :param student_arch_params:        Architecture H.P. e.g.: block, num_blocks, num_classes, etc for student net.
+        :param teacher_arch_params:        Architecture H.P. e.g.: block, num_blocks, num_classes, etc for teacher net.
+        :param teacher_checkpoint_path:    Local path to the teacher's checkpoint. Note that when passing pretrained_weights
+                                           through teacher_arch_params these weights will be overridden by the
+                                           pretrained checkpoint.
+        :param load_kd_model_checkpoint:   Whether to load an entire KDModule checkpoint (used to continue KD training)
+        :param kd_model_source_ckpt_folder_name: Folder name to load an entire KDModule checkpoint from
+            (self.experiment_name if none is given) to resume KD training
+        :param kd_model_external_checkpoint_path: The path to the external checkpoint to be loaded. Can be absolute or relative
+                                           (ie: path/to/checkpoint.pth). If provided, will automatically attempt to
+                                           load the checkpoint even if the load_checkpoint flag is not provided
+        :param freeze_teacher_eval_mode:   Whether to freeze self.teacher at eval mode regardless of self.train(mode)
+        :return:
+        """
+
+        kd_model_arch_params = {}
+
+        if 'num_classes' not in student_arch_params.keys():
+            if self.dataset_interface is None:
+                raise Exception('Error', 'Number of classes not defined in arch params and dataset is not defined')
+            else:
+                student_arch_params['num_classes'] = len(self.classes)
+
+        kd_model_arch_params['num_classes'] = student_arch_params['num_classes']
+        teacher_arch_params['num_classes'] = student_arch_params['num_classes']
+
+        student_arch_params = core_utils.HpmStruct(**student_arch_params)
+        teacher_arch_params = core_utils.HpmStruct(**teacher_arch_params)
+
+        student_net, _ = sg_model_utils.instantiate_net(student_architecture, student_arch_params)
+        teacher_net, _ = sg_model_utils.instantiate_net(teacher_architecture, teacher_arch_params)
+
+        if teacher_checkpoint_path is not None:
```

Comment:
why do you allow teacher_checkpoint_path to be None? I'd think that we should assert if None because then its not distillation